### PR TITLE
Paddle Backend Add features

### PR DESCRIPTION
**Pull request by heliqi**
1. Support native GPU、FP16 and CPU(mkldnn 、ort)
2. Support for more configurations

### CODE REVIEW

File: src/paddle.cc [hunk 1]
```diff
@@ -63,34 +63,45 @@ class ModelImpl {
...
   std::unique_ptr<paddle_infer::Config> analysis_config_;
   std::shared_ptr<paddle_infer::Predictor> predictor_;
-  int device_id_;
+  paddle_infer::PlaceType place_type_;
 };
 
 ModelImpl::ModelImpl(
     const char* model_path, const char* param_path, TRITONPADDLE_Config* config,
-    int device_id)
+    const int32_t device_id)
```
Comment (ZeyuChen) [inline src/paddle.cc:71]: int变量参数不需要加const

File: src/paddle.cc [hunk 2]
```diff
@@ -45,7 +45,7 @@ class ModelImpl {
  public:
   ModelImpl(
       const char* model_path, const char* param_path,
-      TRITONPADDLE_Config* config, int device_id);
+      TRITONPADDLE_Config* config, const int32_t device_id);
```
Comment (ZeyuChen) [inline src/paddle.cc:48]: int变量不需要const

**Review by ZeyuChen — COMMENTED** [review]
确认下其他框架triton集成的device id是否用const，一般来说int变量加const作用不大

File: src/paddle.cc (hunk 1 above)
Comment (heliqi) [reply src/paddle.cc:71]: 这个地方是参照ORT后端的(NV给的文档是以ORT举例子,所以我参考了它):
https://github.com/triton-inference-server/onnxruntime_backend/blob/52b4e6204ad9ced804e43bc1a4a2626c61fef971/src/onnxruntime.cc#L86

**Pull request merged**